chain/processor.go

```diff
@@ -55,10 +55,18 @@ func (b *StatelessBlock) Execute(
 			e.Stop()
 			return nil, nil, err
 		}
-		// Fetch keys from disk or check if it's in cache
-		wg := f.Lookup(ctx, tx.ID(), stateKeys)
+
+		// Prefetch state keys from disk
+		txID := tx.ID()
+		if err := f.Fetch(ctx, txID, stateKeys); err != nil {
```

Comment:
Both these cases require errors to ensure we don't deadlock in the case of a lookup error.